studyBuddy/src/main/java/com/google/sps/models/authmodels/AuthInfo.java

```diff
@@ -0,0 +1,53 @@
+/* 
+This is class models information needed for a user to login and logout of to the site.
+
+Note: The Login Servlet returns a JSON response modeled like this class. 
+
+TODO: Add a method that generates the students's school from their email.       
+*/
+
+package com.google.sps.models.authmodels;
+
+public final class AuthInfo{
+
+    private Boolean isLoggedIn;
+    private String url;
+    private String email;
+
+    // Constructor
+    public AuthInfo(){
+        this.isLoggedIn = false;
+        this.url = "N/A";
+        this.email = "N/A";
+    }
+
+    // Copy constructor
+    public AuthInfo(Boolean isLoggedIn, String url, String email){
```

Comment:
This is more of a question, but from my understanding a copy constructor takes an instance of an object and performs a deep copy. So if this is a copy constructor, then shouldn't it take a single parameter of an AuthInfo instance and then extract from the instance to create a new instance? As it stands, I should be able to create an AuthInfo instance using this constructor with any boolean and two strings, correct?

Either way, this isn't really a functional issue. It is more of just me wanting to make sure I have my vocabulary right.

Reply:
That is right. Maybe I shouldn't have called it Copy Constructor. I am actually going to add some restrictions as to what types of Strings are supposed to be passed and maybe throw and error if they are not met.